avocado/utils/pci.py

```
@@ -376,6 +376,96 @@ def get_vendor_id(full_pci_address):
+    output = get_vendor_id(full_pci_address)
+    vid = output.replace(":", " ")
+
+    # Add device vendor id to the driver
+    try:
+        genio.write_file_or_fail(f"/sys/bus/pci/drivers/{driver}/new_id", f"{vid}\n")
+    except genio.GenIOError as e:
+        if "File exists" in str(e):
+            return
+        raise ValueError(
+            f"Failed to add vendor ID '{vid}' to driver '{driver}': {e}"
+        ) from e
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)